src/main/java/TiTi/task/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package TiTi.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the Deadline task.
+ * Contain String description of the task.
+ * Contain String description of the date of task.
+ * Contain LocalDate object of the date of task (if applicable).
+ * Contain boolean value of whether the task has been completed.
+ */
+public class Deadline extends Task {
+
+    protected String by;
+    protected LocalDate date;
+
+    /**
+     * Constructor for Deadline class.
+     *
+     * @param description description of the task.
+     * @param by string description of the time to complete the task by.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        checkDate(by);
+    }
+
+    private void checkDate(String by) {
```

Comment:
Perhaps this function could be named with a verb, such as `dateChecker`.

Reply:
thanks for the review! but ```dateChecker``` is actually a noun! And by convention class and variable should be names with nouns, and methods should be named with verb! Just saying in case you might have made the same assumptions for your own code ;-;